torchgeo/datasets/glacier_mapping_alps.py

```diff
@@ -0,0 +1,427 @@
+
+    .. versionadded:: 0.7
+    """
+
+    r_url = 'https://huggingface.co/datasets/dcodrut/dl4gam_alps/resolve/7d20ca8a2b30c5518e086ffaa5ce37e6a66c42c1/data'
+    download_metadata: ClassVar[dict[str, dict[str, str]]] = {
+        'dataset_small': {
+            'url': f'{r_url}/patches/inv_r_128_s_128.tar.gz',
+            'checksum': '3e69c47c6ff5106cd4ffaa6bb2caaaef',
+        },
+        'dataset_large': {
+            'url': f'{r_url}/patches/inv_r_128_s_64.tar.gz',
+            'checksum': '06e85a6a9e3dc6b3cdb07f928e832bc8',
+        },
+        'splits_csv': {
+            'url': f'{r_url}/map_all_splits_all_folds.csv',
+            'checksum': '862355c5c3482271dd171d31c70551b3',
+        },
+    }
+
+    rgb_bands = ('B4', 'B3', 'B2')
+    all_bands = (
+        'B1',
+        'B2',
+        'B3',
+        'B4',
+        'B5',
+        'B6',
+        'B7',
+        'B8',
+        'B8A',
+        'B9',
+        'B10',
+        'B11',
+        'B12',
+    )
+    rgb_nir_swir_bands = ('B4', 'B3', 'B2', 'B8', 'B11')  # the subset used in our paper
+
+    valid_extra_features = (
+        'dem',  # Digital Elevation Model
+        'slope',
+        'aspect',
+        'planform_curvature',
+        'profile_curvature',
+        'terrain_ruggedness_index',  # DEM-based features
+        'dhdt',  # surface elevation change
+        'v',  # surface velocity
+    )
+    valid_splits = ('train', 'val', 'test')
+    valid_versions = ('small', 'large')
+    valid_cv_iters = (1, 2, 3, 4, 5)
+
+    def __init__(
+        self,
+        root: str | os.PathLike[str] = 'data',
+        split: str = 'train',
+        cv_iter: int = 1,
+        version: str = 'small',
+        bands: Sequence[str] = rgb_nir_swir_bands,
+        extra_features: Sequence[str] | None = None,
+        transforms: Callable[[dict[str, Tensor]], dict[str, Tensor]] | None = None,
+        download: bool = False,
+        checksum: bool = False,
+    ) -> None:
+        """Initialize the dataset.
+
+        Args:
+            root: root directory where dataset can be found
+            split: one of "train", "val", or "test"
+            cv_iter: one of 1, 2, 3, 4, 5 (for the five-fold geographical cross-validation scheme)
+            version: one of "small" or "large" (controls the sampling overlap)
+            bands: the Sentinel-2 bands to use as input (default: RGB + NIR + SWIR)
+            extra_features: additional features to include (default: None; see the class attribute for the available)
+            transforms: a function/transform that takes input sample and its target as entry and returns a transformed
+             version
+            download: if True, download dataset and store it in the root directory
+            checksum: if True, check the MD5 of the downloaded files (may be slow)
+
+        Raises:
+            AssertionError: if the ``split``, ``cv_iter``, ``version``, ``bands`` or ``extra_features`` are invalid
```

Comment:
I usually use `*foo*` instead of backticks because that's what the Python docs uses. Could also be less specific and just say "if any parameters are invalid".